app/src/main/cpp/skyline/gpu/interconnect/maxwell_3d/pipeline_manager.cpp

```diff
@@ -942,12 +943,20 @@ namespace skyline::gpu::interconnect::maxwell3d {
         });
     }
 
-    PipelineManager::PipelineManager(GPU &gpu) {
+    PipelineManager::PipelineManager(GPU &gpu, JvmManager &jvm) {
         if (!gpu.graphicsPipelineCacheManager)
             return;
 
-        std::ifstream stream{gpu.graphicsPipelineCacheManager->OpenReadStream()};
+        u32 pipelineCount{};
+        jint compiledCount{};
```

Comment:
Don't use java types in GPU code, this also needs to be atomic. 

std::atomic<i32> currentCompiledPipelineCount{};

Reply:
Does it really need to be atomic? It's an 32bit int, it's going to be atomic in any case, and even if it isn't we don't really care, do we?

Reply:
I made it atomic

Reply:
> Does it really need to be atomic? It's an 32bit int, it's going to be atomic in any case, and even if it isn't we don't really care, do we?

We aren't x86, memory accesses are not generally atomic on arm